rust/src/nasl/builtin/ssh/libssh/mod.rs

```diff
@@ -0,0 +1,152 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
+
+//! Defines NASL ssh and sftp functions
+
+mod channel;
+mod session;
+
+use std::net::{IpAddr, UdpSocket};
+use std::{os::fd::AsRawFd, time::Duration};
+
+use libssh_rs::{LogLevel, SshOption};
+use russh::cipher;
+use russh_keys::key;
+use tokio::sync::{Mutex, MutexGuard};
+use tracing::debug;
+
+use super::error::Result;
+use super::Ssh;
+
+pub use libssh_rs::AuthMethods;
+pub use session::SshSession;
+
+pub type SessionId = i32;
+pub type Socket = std::os::raw::c_int;
+
+pub fn get_log_level() -> LogLevel {
+    let verbose = std::env::var("OPENVAS_LIBSSH_DEBUG")
```

Comment:
q: what's the reasoning to not follow to overall logging level?

Reply:
I don't know. I did this because it was implemented in the [original implementation](https://github.com/greenbone/openvas-scanner/blob/e6abbc2a30283a3f036ba3ccf24bb8a82ac3354e/rust/src/nasl/builtin/ssh/mod.rs#L467-L469).

Reply:
This is a little more annoying than I would like - as far as I can tell there is no way to retrieve the active level filter from `tracing_subscriber`, so we'll have to remember the logging level and store it in the context, for example. I'm not sure this is what we want to do.

Reply:
Maybe https://docs.rs/tracing/latest/tracing/level_filters/struct.LevelFilter.html#method.current helps

Reply:
Maybe `tracing_subscriber::filter::LevelFilter::current() ` will work